src/app/app.component.ts

```diff
@@ -854,36 +907,50 @@ export class AppComponent implements OnInit, AfterViewInit, CanvasTableSelectLis
 
   dropToFolder(folderId): void {
     const messageIds = this.canvastable.rows.selectedMessageIds();
+    let folderPath;
+    this.messagelistservice.folderListSubject.subscribe(folders => {
+      folderPath = folders.find(fld => fld.folderId === folderId).folderPath;
```

Comment:
I don't think we can count on this actually being executed synchronously – to be safe, everything that uses `folderPath` below should be a part of this `subscribe()` callback.